bson/bsonrw/extjson_writer.go

```diff
@@ -572,6 +572,10 @@ func (ejvw *extJSONValueWriter) WriteDocumentEnd() error {
 	case mDocument:
 		ejvw.buf = append(ejvw.buf, ',')
 	case mTopLevel:
+		// Always end top-level documents with a newline so that multiple documents can be encoded
+		// to the same writer. That matches the Go json.Encoder behavior and also works with
+		// bsonrw.NewExtJSONValueReader.
+		ejvw.buf = append(ejvw.buf, '\n')
```

Comment:
This addition causes [corpus failures](https://spruce.mongodb.com/task/mongo_go_driver_tests_42_plus_zlib_zstd_support__version~latest_os_ssl_40~ubuntu1804_64_go_1_19_test_standalone_noauth_nossl_patch_54a3206c25f4b625c43aedbed95d83123d8bb7f7_645957d15623436242ec6cde_23_05_08_20_13_06/tests?execution=0&sortBy=STATUS&sortDir=ASC): 

```
go test -run Test_BsonCorpus
```

Reply:
Good catch! To match the behavior of the Go `encoding/json` library, we actually only want to apply this to the output generated by `Encode` and not by `MarshalExtJSON`. I'm not sure if the BSON corpus tests use `Encode` or `MarshalExtJSON`, but it's possible that change will resolve the failures. If not, I'll look into whether or not it's a good idea to trim whitespace in the BSON corpus tests.